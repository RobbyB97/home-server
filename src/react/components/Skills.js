import React from 'react';


export class Skills extends React.Component {
    render() {
        return (
            <section id="Skills" className="skills">
                Skills component
            </section>
        );
    }
}


export default Skills;