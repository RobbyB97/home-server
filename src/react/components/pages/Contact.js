import React from 'react'
import {connect} from 'react-redux'



const Contact = ({dispatch, nav, title}) => (
	<div id="contact">
		Contact component
	</div>
)



const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav,
		title: state.title.title,
	}
}

export default connect(mapStateToProps)(Contact)