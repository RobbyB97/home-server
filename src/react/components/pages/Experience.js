import React from 'react'
import {connect} from 'react-redux'

import Job from '../content/Job'
import PrevPage from '../common/PrevPage'
import NextPage from '../common/NextPage'


export class Experience extends React.Component {
    
    render() {
        return (
            <div id="experience">
                
                Experience component
                <Job
                    title="Finalsite"
                    text="Job"
                    image="/dist/images/icons/finalsite.png">
                </Job>

                <PrevPage
                    text="Skills"
                    route="/skills/"
                    float={true}
                />

                <NextPage 
                    text="Projects"
                    route="/projects/"
                    float={true}
                />
            </div>  
        )
    }
}


const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav,
        title: state.title.title,
        splash: state.home.splash
	}
}

const mapDispatchToProps = (dispatch) => ({

})


export default connect(mapStateToProps, mapDispatchToProps)(Experience)