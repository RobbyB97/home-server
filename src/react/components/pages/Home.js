import React from 'react'
import {connect} from 'react-redux'

import Attribute from '../content/Attribute'


export class Home extends React.Component {

    render() {

        const fastAttribute = {
            image: "/dist/images/icons/speed_icon.png",
            title: "Fast",
            text: "Fully minified files and an optimized structure guarantee your website will load quickly, even on inconsistent internet connections."
        }

        const responsiveAttribute = {
            image: "/dist/images/icons/responsive_icon.png",
            title: "Responsive",
            text: "Mobile first development built up to 4k screens guarantees that your website will be elegant and functional on any device."
        }

        const beautifulAttribute = {
            image: "/dist/images/icons/paint_icon.png",
            title: "Beautiful",
            text: "A thorough, UX focused design process and open communication guarantees your website will look as beautiful as you envision it."
        }

        const accessibleAttribute = {
            image: "/dist/images/icons/accessibility_icon.png",
            title: "Accessible",
            text: "Full compliance with ADA standards guarantees that your website will be accessible and easy to use for everyone."
        }
        
        return (
            <div id="home">
                
                <div className={this.props.splash ? "home__background phone one": "home__background phone two"}></div>
                <div className={this.props.splash ? "home__background tablet one": "home__background tablet two"}></div>
                <div className={this.props.splash ? "home__background desktop one": "home__background desktop two"}></div>
                <div className={this.props.splash ? "home__background huge one": "home__background huge two"}></div>

                {!this.props.splash &&
                    <div className="home__attributes">
                        <Attribute {...fastAttribute} />
                        <Attribute {...responsiveAttribute} />
                        <Attribute {...beautifulAttribute} />
                        <Attribute {...accessibleAttribute} />
                    </div>
                }
                
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
      nav: state.navigation.nav,
      title: state.title.title,
      home: state.home.home,
      homePage: state.home.homePage,
      splash: state.home.splash
    }
}

const mapDispatchToProps = (dispatch) => ({

})


export default connect(mapStateToProps, mapDispatchToProps)(Home)