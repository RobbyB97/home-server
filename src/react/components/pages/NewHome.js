import React from 'react'


export class NewHome extends React.Component {
    render() {
        return (
            <div id="newHome">                
                <div className="newHome__content">
                    <div className="newHome__head">
                        <div className="newHome__name">
                            <h2>Robby Bergers</h2>
                        </div>
                        <div className="newHome__title">
                            <h3>Front End Developer</h3>
                        </div>
                    </div>

                    <div className="newHome__body">

                    </div>
                </div>

                <div className="newHome__backdrop"></div>
            </div>
        )
    }
}


export default NewHome