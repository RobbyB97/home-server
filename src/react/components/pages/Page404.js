import React from 'react'
import {NavLink} from 'react-router-dom'


export class Page404 extends React.Component {
    
    render() {
        return (
            <div id="page404">
                <div className="page404__backdrop"></div>
                <div className="page404__content">
                    <div className="page404__image">
                        <img src="/dist/images/wildcard.png" />
                    </div>

                    <div className="page404__text">
                        Uh oh, looks like this page doesn't exist.
                    </div>

                    <div className="page404__home-button">
                        <NavLink to="/">
                            <button>
                                Return home
                            </button>
                        </NavLink>
                    </div>
                </div>
            </div>
        )
    }
}


export default Page404