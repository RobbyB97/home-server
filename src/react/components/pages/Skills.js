import React from 'react'
import {connect} from 'react-redux'

import Skill from '../content/Skill'
import NextPage from '../common/NextPage'
import PrevPage from '../common/PrevPage'


export class Skills extends React.Component {
    
    render() {
        return (
            <div id="skills">
                Skills component

                <PrevPage
                    text="Homepage"
                    route="/"
                    float={true}
                />

                <NextPage 
                    text="Experience"
                    route="/experience/"
                    float={true}
                />
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        nav: state.navigation.nav,
        title: state.title.title,
    }
}

const mapDispatchToProps = (dispatch) => ({

})


export default connect(mapStateToProps, mapDispatchToProps)(Skills)