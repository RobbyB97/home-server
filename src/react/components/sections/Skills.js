import React from 'react';

import SkillSection from '../SkillSection';


export class Skills extends React.Component {
    render() {
        const sections = [
            {
                title: "Front end",
                skills: [
                    "React", "SASS", "HTML5", "CSS3", "JS (ES6)"
                ]
            }, {
                title: "Workflow",
                skills: [
                    "Git", "NPM", "Gulp", "Webpack", "Jest", 
                    "Firebase", "Heroku"
                ]
            }, {
                title: "Back end",
                skills: [
                    "Node.js", "BASH", "Linux/Ubuntu server",
                    "MongoDB", "Oracle SQL"
                ]
            }
        ];

        return (
            <section id="Skills" className="skills">
                <header className="skills__header">
                    <h2>My skillset</h2>
                </header>

                {sections.map((section) => (
                    <SkillSection 
                        {...section}
                        key={section.title}
                    />
                ))}
            </section>
        );
    }
}


export default Skills;