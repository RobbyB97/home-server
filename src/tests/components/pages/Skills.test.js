import React from 'react'
import {shallow} from 'enzyme'

import {Skills} from '../../../react/components/pages/Skills'


let wrapper

beforeEach(() => {
    wrapper = shallow(<Skills />)
})


test('Render Skills page', () => {
    expect(wrapper).toMatchSnapshot()
})