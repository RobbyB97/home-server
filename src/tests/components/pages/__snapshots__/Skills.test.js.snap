// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Skills page 1`] = `
<div
  id="skills"
>
  Skills component
  <PrevPage
    float={true}
    route="/"
    text="Homepage"
  />
  <NextPage
    float={true}
    route="/experience/"
    text="Experience"
  />
</div>
`;
